stl/inc/__msvc_string_view.hpp

```diff
@@ -817,48 +843,31 @@ constexpr size_t _Traits_find_first_of(_In_reads_(_Hay_size) const _Traits_ptr_t
     const auto _Hay_end   = _Haystack + _Hay_size;
 
     if constexpr (_Is_implementation_handled_char_traits<_Traits>) {
-        if (!_STD _Is_constant_evaluated()) {
-            using _Elem = typename _Traits::char_type;
-
 #if _USE_STD_VECTOR_ALGORITHMS
-            const bool _Try_vectorize = _Hay_size - _Start_at > _Threshold_find_first_of;
-
-            // Additional condition for when the vectorization outperforms the table lookup
-            constexpr size_t _Find_first_of_bitmap_threshold = sizeof(_Elem) == 1 ? 48 : sizeof(_Elem) == 8 ? 8 : 16;
-
-            const bool _Use_bitmap = !_Try_vectorize || _Needle_size > _Find_first_of_bitmap_threshold;
-#else // ^^^ _USE_STD_VECTOR_ALGORITHMS / !_USE_STD_VECTOR_ALGORITHMS vvv
-            const bool _Use_bitmap = true;
-#endif // ^^^ !_USE_STD_VECTOR_ALGORITHMS ^^^
-
-            if (_Use_bitmap) {
-                _String_bitmap<_Elem> _Matches;
-
-                if (_Matches._Mark(_Needle, _Needle + _Needle_size)) {
-                    for (auto _Match_try = _Hay_start; _Match_try < _Hay_end; ++_Match_try) {
-                        if (_Matches._Match(*_Match_try)) {
-                            return static_cast<size_t>(_Match_try - _Haystack); // found a match
-                        }
-                    }
-                    return static_cast<size_t>(-1); // no match
+        if (!_STD _Is_constant_evaluated()) {
+            const size_t _Remaining_size = _Hay_size - _Start_at;
+            if (_Remaining_size + _Needle_size >= _Threshold_find_first_of) {
+                size_t _Pos = _Find_first_of_pos_vectorized(_Hay_start, _Remaining_size, _Needle, _Needle_size);
+                if (_Pos != static_cast<size_t>(-1)) {
+                    _Pos += _Start_at;
                 }
-
-                // couldn't put one of the characters into the bitmap, fall back to vectorized or serial algorithms
+                return _Pos;
             }
+        }
+#endif // _USE_STD_VECTOR_ALGORITHMS
 
-#if _USE_STD_VECTOR_ALGORITHMS
-            if (_Try_vectorize) {
-                const _Traits_ptr_t<_Traits> _Found =
-                    _STD _Find_first_of_vectorized(_Hay_start, _Hay_end, _Needle, _Needle + _Needle_size);

```

Comment:
The dispatcher `_Find_first_of_vectorized` is declared in `<xutility>` but called only in `<algorithm>` now; it should be moved down.